usaspending_api/agency/v2/views/budget_function_count.py

```diff
@@ -0,0 +1,47 @@
+from django.db.models import Exists, OuterRef
+from rest_framework.request import Request
+from rest_framework.response import Response
+from typing import Any
+from usaspending_api.agency.v2.views.agency_base import AgencyBase
+from usaspending_api.common.cache_decorator import cache_response
+from usaspending_api.financial_activities.models import FinancialAccountsByProgramActivityObjectClass
+from usaspending_api.accounts.models import TreasuryAppropriationAccount
+
+
+class BudgetFunctionCount(AgencyBase):
+    """
+    Obtain the count of budget functions for a specific agency in a single
+    fiscal year based on whether or not that budget function has ever
+    been submitted in File B.
+    """
+
+    endpoint_doc = "usaspending_api/api_contracts/contracts/v2/agency/toptier_code/budget_function/count.md"
+
+    @cache_response()
+    def get(self, request: Request, *args: Any, **kwargs: Any) -> Response:
+        rows = list(self.budget_function_queryset())
+
+        return Response(
+            {
+                "toptier_code": self.toptier_code,
+                "fiscal_year": self.fiscal_year,
+                "budget_function_count": len(set([row["budget_function_code"] for row in rows])),
+                "budget_sub_function_count": len(set([row["budget_subfunction_code"] for row in rows])),
+            }
+        )
+
+    def budget_function_queryset(self):
```

Comment:
nitpicking, but is there a difference between a query and a queryset? This kind of function was just called `query` before.

Reply:
Before?
Queries are SQL and run in a database. Django ORM has querysets to abstract SQL from the business logic.

Reply:
I thought the object class endpoint called it `object_class_query`, but I see that that's not the case. If the return value is a queryset (or could be abstracted to one), then I have no complaints.

Reply:
Thankfully I did code it to return a queryset. Whew, glad I avoided that conflict